### PR TITLE
🔥 Impement Delta Live Tables support

### PR DESCRIPTION
**Pull request by renardeinside**
## Proposed changes

Following the #522 , this PR re-introduces DLT and adds this functionality to dbx.
Previous #451 closed since too much changes were brought by #522 .

Original issue - #250 

## Types of changes

What types of changes does your code introduce to dbx?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: docs/guides/general/delta_live_tables.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      storage: "dbfs:/some/location" #(2)
+      configuration: #(3)
+        "property1": "value"
+        "property2": "value2"
+      clusters: #(4)
+        - label: "some-label" #(5)
+          spark_conf:
+            "spark.property1": "value"
+            "spark.property2": "value2"
+            aws_attributes:
```
Comment (ctaylor-carvana) [inline docs/guides/general/delta_live_tables.md:29]: Is this the intended nesting? Would it be possible to include an example of enhanced autoscaling definition?

File: dbx/models/workflow/common/workflow_types.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class WorkflowType(str, Enum):
```
[the remark this answers is not in the document]
Comment (renardeinside) [reply dbx/models/workflow/common/workflow_types.py:4]: I haven't yet finished, it's just the logical modeling part :)

File: docs/guides/general/delta_live_tables.md (hunk 1 above)
Comment (renardeinside) [reply docs/guides/general/delta_live_tables.md]: please check the updated version

File: docs/guides/general/delta_live_tables.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        - label: "some-label" #(5)
+          spark_conf:
+            "spark.property1": "value"
+            "spark.property2": "value2"
+          aws_attributes:
+            ...
+          instance_pool_id: "instance-pool://some-pool"
+          driver_instance_pool_id: "instance-pool://some-pool"
+          policy_id: "cluster-policy://some-policy"
+          autoscale:
```
Comment (ctaylor-carvana) [inline docs/guides/general/delta_live_tables.md:34]: Hey! I tried with the new example; everything else worked perfectly except the autoscale. Curious why?

ValidationError: 1 validation error for Deployment
workflows -> 0 -> Pipeline -> clusters -> 0 -> autoscale
  Model values must be a dict; you may not have returned a dictionary from a root validator (type=type_error)

Comment (renardeinside) [reply docs/guides/general/delta_live_tables.md:34]: fixed in the later commits, thanks!

**Pull request merged**